src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.cpp

```diff
@@ -40,104 +36,406 @@ namespace fastdds {
+    for (const xtypes::TypeIdentifier& type_id : request.type_ids())
+    {
+        // If TypeIdentifier is EK_MINIMAL add complete_to_minimal to answer
+        if (type_id._d() == xtypes::EK_MINIMAL)
+        {
+            minimal_id = type_id;
+            // Get complete TypeIdentifier from registry
+            complete_id = fastrtps::rtps::RTPSDomainImpl::get_instance()->type_object_registry_observer().
+                            get_complementary_type_identifier(minimal_id);
 
-                typelookup_manager_->send_reply(*reply);
-                typelookup_manager_->reply_type_.deleteData(reply);
+            xtypes::TypeIdentifierPair id_pair;
+            id_pair.type_identifier1(complete_id);
+            id_pair.type_identifier2(minimal_id);
+            // Add the id pair to the result
+            out.complete_to_minimal().push_back(std::move(id_pair));
+        }
+        else
+        {
+            complete_id = type_id;
+        }
 
-                break;
+        type_result = fastrtps::rtps::RTPSDomainImpl::get_instance()->type_object_registry_observer().
+                        get_type_object(complete_id, obj);
+        if (RETCODE_OK != type_result)
+        {
+            // If any object is unknown, abort and answer with exception
+            break;
+        }
+
+        xtypes::TypeIdentifierTypeObjectPair id_obj_pair;
+        id_obj_pair.type_identifier(complete_id);
+        id_obj_pair.type_object(obj);
+        // Add the id/obj pair to the result
+        out.types().push_back(std::move(id_obj_pair));
+    }
+
+    if (RETCODE_OK == type_result)
+    {
+        // Prepare and send the reply
+        answer_request(request_id, rpc::RemoteExceptionCode_t::REMOTE_EX_OK, out);
+    }
+    else
+    {
+        // If any of the types is not found, log error
+        EPROSIMA_LOG_WARNING(TYPELOOKUP_SERVICE_REQUEST_LISTENER, "Error getting type.");
+        answer_request(request_id, rpc::RemoteExceptionCode_t::REMOTE_EX_INVALID_ARGUMENT);
+    }
+}
+
+void TypeLookupRequestListener::check_get_type_dependencies_request(
+        SampleIdentity request_id,
+        const TypeLookup_getTypeDependencies_In& request)
+{
+    std::unordered_set<xtypes::TypeIdentfierWithSize> type_dependencies;
+    ReturnCode_t type_dependencies_result = RETCODE_ERROR;
+    if (!request.type_ids().empty())
+    {
+        // Check if the received request has been done before and needed a continuation point
+        std::lock_guard<std::mutex> lock(requests_with_continuation_mutex_);
+        if (!request.continuation_point().empty())
+        {
+            auto requests_it = requests_with_continuation_.find(request.type_ids());
+            if (requests_it != requests_with_continuation_.end())
+            {
+                // Get the dependencies without chechking the registry
+                type_dependencies = requests_it->second;
+                type_dependencies_result = RETCODE_OK;
             }
```

Comment:
What happens if it is not found in the collection? It should not happen, but it should be considered.

Reply:
added better error check in [d054d39](https://github.com/eProsima/Fast-DDS/pull/4240/commits/d054d39484489f3890638c8cfe4fe4c023139c74)